File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def get_num(n):
    count = 0
    while (n >= 50 and n % 2 == 0) or (n < 50 and n % 2 != 0):
        if n >= 50 and n % 2 == 0:
            n //= 2
        else:
            n = n * 2 + 1
        count += 1
    
    return count

def solution(arr):
    dp = {}
    for i in range(101):
        dp[i] = get_num(i)
    
    answer = []
    for i in range(len(arr)):
        answer.append(dp[arr[i]])

    return max(answer)

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.*;

class Solution {
    public int getNum(int n) {
        int count = 0;
        while ((n >= 50 && n % 2 == 0)||(n < 50 && n % 2 != 0)) {
                if (n >= 50 && n % 2 == 0) {
                    n /= 2;
                } else if (n < 50 && n % 2 != 0) {
                    n = 2 * n + 1; 
                }
            count += 1;
            }
        return count;
    }
    
    public int solution(int[] arr) {
        int answer = 0;
        for (int i : arr) {
            answer = Math.max(answer, getNum(i));
        }
        
        return answer;
    }
}